syntax = "proto3";

message DemoCreateRequest {
  string name = 1;
}

message DemoDeleteRequest {
  string id = 1;
}

message DemoUpdateRequest {
  string id = 1;
  string name = 2;
}

message DemoQueryRequest {
  string id = 1;
}

message DemoListRequest {
  int64 latestCreateTime = 1;
}

message DemoCommonResponse {
  string id = 1;
  string name = 2;
  int64 createTime = 3;
  int64 updateTime = 4;
}

message DemoListResponse {
  repeated DemoCommonResponse demos = 1;
}
